boards/nxp/mimxrt1180_evk/board.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2024 NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/devicetree.h>
+#include <soc.h>
+#if defined(CONFIG_SOC_MIMXRT1189_CM7)
+#include <zephyr/cache.h>
+#elif defined(CONFIG_IMXRT118X_CM33_XCACHE_PS)
+#include <fsl_cache.h>
+#endif
+
+static int board_init_netc(void)
+{
+#if defined(CONFIG_ETH_NXP_IMX_NETC) && (DT_CHILD_NUM_STATUS_OKAY(DT_NODELABEL(netc)) != 0)
+/* TODO: support NETC with cache enablement */
+#if defined(CONFIG_IMXRT118X_CM33_XCACHE_PS)
+	XCACHE_DisableCache(XCACHE_PS);
+	XCACHE_DisableCache(XCACHE_PC);
+#elif defined(CONFIG_SOC_MIMXRT1189_CM7)
+	sys_cache_data_disable();
+	sys_cache_instr_disable();
+#endif
+	/* RMII mode */
+	BLK_CTRL_WAKEUPMIX->NETC_LINK_CFG[0] = BLK_CTRL_WAKEUPMIX_NETC_LINK_CFG_MII_PROT(1);
+	BLK_CTRL_WAKEUPMIX->NETC_LINK_CFG[4] = BLK_CTRL_WAKEUPMIX_NETC_LINK_CFG_MII_PROT(1);
+
+	/* Output reference clock for RMII */
+	BLK_CTRL_WAKEUPMIX->NETC_PORT_MISC_CFG |=
+		BLK_CTRL_WAKEUPMIX_NETC_PORT_MISC_CFG_PORT0_RMII_REF_CLK_DIR_MASK;
+	BLK_CTRL_WAKEUPMIX->NETC_PORT_MISC_CFG |=
+		BLK_CTRL_WAKEUPMIX_NETC_PORT_MISC_CFG_PORT4_RMII_REF_CLK_DIR_MASK;
+
+	/* Unlock the IERB. It will warm reset whole NETC. */
+	NETC_PRIV->NETCRR &= ~NETC_PRIV_NETCRR_LOCK_MASK;
+	while ((NETC_PRIV->NETCRR & NETC_PRIV_NETCRR_LOCK_MASK) != 0U) {
+	}
+
+	/* Set the access attribute, otherwise MSIX access will be blocked. */
+	NETC_IERB->ARRAY_NUM_RC[0].RCMSIAMQR &= ~(7U << 27);
+	NETC_IERB->ARRAY_NUM_RC[0].RCMSIAMQR |= (1U << 27);
+
+	/* Lock the IERB. */
+	NETC_PRIV->NETCRR |= NETC_PRIV_NETCRR_LOCK_MASK;
+	while ((NETC_PRIV->NETCSR & NETC_PRIV_NETCSR_STATE_MASK) != 0U) {
+	}
+#endif
+	return 0;
+}
+
+SYS_INIT(board_init_netc, POST_KERNEL, CONFIG_BOARD_MIMXRT1180_EVK_NETC_INIT_PRIO);
```

Comment:
use board init hook (new feature recently merged)

Reply:
okay, thanks.

Reply:
Just checked the new added hooks. Looks like both board_early_init_hook and board_late_init_hook don't suit.
These netc init code should be run after clock init which is PRE_KERNEL_1 level, and before NETC driver init which is POST_KERNEL level.
But board_early_init_hook is too early before PRE_KERNEL_1, and board_late_init_hook is too late after POST_KERNEL.

Reply:
the soc clock init should be running as part of the soc init hook now ?

Reply:
No... Let me convert to that.
Thanks.